### ISSUE TITLE
Upgraded Boot2Docker to 1.3.0 with Docker 1.3.0 now I can't connect my Docker Cloud.

### ISSUE THREAD

**Issue by mtscout6** [jenkinsci/docker-plugin#108]
I updated Boot2Docker to 1.3.0 today which uses Docker 1.3.0 and I believe that the connection is now secured. When I upgraded I had to update my ENV from 

```
DOCKER_HOST=tcp://192.168.59.103:2375
```

to

```
DOCKER_HOST=tcp://192.168.59.103:2376
DOCKER_CERT_PATH=/Users/smithm/.boot2docker/certs/boot2docker-vm
DOCKER_TLX_VERIFY=1
```

I thus changed the Docker URL in the Jenkins system configuration from `http://192.168.59.103:2375` to `http://192.168.59.103:2376` Which yields the error:

```
javax.servlet.ServletException: javax.ws.rs.ProcessingException: java.net.SocketException: Unexpected end of file from server
    at org.kohsuke.stapler.Stapler.tryInvoke(Stapler.java:783)
    at org.kohsuke.stapler.Stapler.invoke(Stapler.java:863)
    at org.kohsuke.stapler.MetaClass$6.doDispatch(MetaClass.java:248)
    at org.kohsuke.stapler.NameBasedDispatcher.dispatch(NameBasedDispatcher.java:53)
    at org.kohsuke.stapler.Stapler.tryInvoke(Stapler.java:733)
    at org.kohsuke.stapler.Stapler.invoke(Stapler.java:863)
    at org.kohsuke.stapler.Stapler.invoke(Stapler.java:636)
    at org.kohsuke.stapler.Stapler.service(Stapler.java:225)
    at javax.servlet.http.HttpServlet.service(HttpServlet.java:848)
    at org.eclipse.jetty.servlet.ServletHolder.handle(ServletHolder.java:686)
    at org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1494)
    at hudson.util.PluginServletFilter$1.doFilter(PluginServletFilter.java:96)
...
```

When I tried `https://192.168.59.103:2376` I got:

```
javax.servlet.ServletException: javax.ws.rs.ProcessingException: javax.net.ssl.SSLHandshakeException: sun.security.validator.ValidatorException: PKIX path building failed: sun.security.provider.certpath.SunCertPathBuilderException: unable to find valid certification path to requested target
    at org.kohsuke.stapler.Stapler.tryInvoke(Stapler.java:783)
    at org.kohsuke.stapler.Stapler.invoke(Stapler.java:863)
    at org.kohsuke.stapler.MetaClass$6.doDispatch(MetaClass.java:248)
    at org.kohsuke.stapler.NameBasedDispatcher.dispatch(NameBasedDispatcher.java:53)
    at org.kohsuke.stapler.Stapler.tryInvoke(Stapler.java:733)
    at org.kohsuke.stapler.Stapler.invoke(Stapler.java:863)
    at org.kohsuke.stapler.Stapler.invoke(Stapler.java:636)
    at org.kohsuke.stapler.Stapler.service(Stapler.java:225)
    at javax.servlet.http.HttpServlet.service(HttpServlet.java:848)
    at org.eclipse.jetty.servlet.ServletHolder.handle(ServletHolder.java:686)
    at org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1494)
    at hudson.util.PluginServletFilter$1.doFilter(PluginServletFilter.java:96)
    at hudson.util.PluginServletFilter.doFilter(PluginServletFilter.java:88)
    at org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1482)
    at hudson.security.csrf.CrumbFilter.doFilter(CrumbFilter.java:48)
    at org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1482)
    at hudson.security.ChainedServletFilter$1.doFilter(ChainedServletFilter.java:84)
    at hudson.security.ChainedServletFilter.doFilter(ChainedServletFilter.java:76)
    at hudson.security.HudsonFilter.doFilter(HudsonFilter.java:164)
    at org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1482)
    at org.kohsuke.stapler.compression.CompressionFilter.doFilter(CompressionFilter.java:46)
    at org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1482)
...
```

Has anyone else ran into this? Do I need to add some environment variables to my Jenkins server to identify the location of these certs? (This is my best guess on what's missing)

**Comment by JeremyMarshall** [comment]
Forgive my late entry but I was trying to use the docker plugin on my mac but couldn't get Jenkins to connect to my Docker

I did the brew install of docker and boot2docker and got a tcp: port but Jenkins will not connect when I try the test in the global config

Should I pull the latest master and build or am I missing a vital step?

**Comment by thomassuckow** [comment]
You probably need to use the 0.9-SNAPSHOT and configure Jenkins with the keys for authenticating to docker.

**Comment by JeremyMarshall** [comment]
I used this one http://repo.jenkins-ci.org/snapshots/com/nirima/docker-plugin/0.9-SNAPSHOT/ and it worked without swapping keys 

Thanks

**Comment by Downchuck** [comment]
What is the approximate command to make the PEM; also, is there a method for removing uploaded certs?

This did not work out for me:
```
openssl pkcs12 -export -out cert.pfx -inkey /var/lib/boot2docker/tls/key.pem -in /var/lib/boot2docker/tls/cert.pem -certfile /var/lib
/boot2docker/tls/ca.pem
```

**Comment by KostyaSha** [comment]
In summary: what is this issue about?

**Comment by jmiddleton** [comment]
Matt, I'm trying to configure the plugin but when I set https://192.168.59.103:2376, I got shaded.org.apache.http.conn.UnsupportedSchemeException: https protocol is not supported.

I'm using Docker Plugin 0.92 with Docker 1.3.0. I have Jenkins server on Kitematic (Virtualbox) and another VM with Docker 1.3.0

Do I need to copy the ....boot2docker/certs/boot2docker-vm/*.pem to Jenkins container?

Thanks,
Jorge
